BUG: _calc_uniform_order_statistic_medians documentation example disagrees with required import

### Describe your issue.

The documentation on how to use the  _calc_uniform_order_statistic_medians function [states here in the example:](https://github.com/scipy/scipy/blob/59d974bf14ff8c82f3353e2d2a9282d9bda67db5/scipy/stats/_morestats.py#L410):


However, _calc_uniform_order_statistic_medians is not exported into the `scipy.stats` package, [it does not appear in any __init__.py at all](https://github.com/scipy/scipy/search?q=_calc_uniform_order_statistic_medians).


### Reproducing Code Example

```python
    >>> from scipy.stats import _calc_uniform_order_statistic_medians
    >>> _calc_uniform_order_statistic_medians(n)
    array([0.15910358, 0.38545246, 0.61454754, 0.84089642])
```


### Error message

```shell
E   ImportError: cannot import name '_calc_uniform_order_statistic_medians' from 'scipy.stats' (/opt/anaconda3/envs/py310/lib/python3.10/site-packages/scipy/stats/__init__.py)
```

### Potential Fix

I would be happy to make a PR for this, but I'm unsure if we should update the documentation to reflect the correct import::

```python
    >>> from scipy.stats._morestats import _calc_uniform_order_statistic_medians
    >>> _calc_uniform_order_statistic_medians(n)
    array([0.15910358, 0.38545246, 0.61454754, 0.84089642])
```

Or if we should add the import into the `__init__.py` in `scipy.stats` so the code from the documentation works.



### SciPy/NumPy/Python version information

1.8.0 1.21.5 sys.version_info(major=3, minor=10, micro=3, releaselevel='final', serial=0)

--- a/scipy/stats/_morestats.py
+++ b/scipy/stats/_morestats.py
@@ -407,7 +407,7 @@ def _calc_uniform_order_statistic_medians(n):
     and greatest order statistics, and the remaining medians are approximated
     by points spread evenly across a sub-interval of the unit interval:
 
-    >>> from scipy.stats import _calc_uniform_order_statistic_medians
+    >>> from scipy.stats._morestats import _calc_uniform_order_statistic_medians
     >>> _calc_uniform_order_statistic_medians(n)
     array([0.15910358, 0.38545246, 0.61454754, 0.84089642])
 

